infra/networking.tf

```diff
@@ -0,0 +1,77 @@
+locals {
+  subnet_config = [
+    {
+      cidr_block        = "172.31.16.0/20"
+      availability_zone = "eu-west-2a"
+    },
+    {
+      cidr_block        = "172.31.32.0/20"
+      availability_zone = "eu-west-2b"
+    },
+    {
+      cidr_block        = "172.31.0.0/20"
+      availability_zone = "eu-west-2c"
+    }
+  ]
+  environment = var.environment == "non-prod" ? "dev" : var.environment
```

Comment:
I think it's going to get confusing having `local.environment` and `var.environment` with different values. Could we pick one and use it everywhere?